src/Application/Calculator/ProductVariantLowestPriceCalculator.php

```diff
@@ -11,24 +11,32 @@
 
 declare(strict_types=1);
 
-namespace Sylius\PriceHistoryPlugin\Application\Provider;
+namespace Sylius\PriceHistoryPlugin\Application\Calculator;
 
 use Sylius\Component\Core\Exception\MissingChannelConfigurationException;
-use Sylius\Component\Core\Model\ChannelInterface;
 use Sylius\Component\Core\Model\ProductVariantInterface;
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelInterface;
 use Sylius\PriceHistoryPlugin\Domain\Model\ChannelPricingInterface;
+use Webmozart\Assert\Assert;
 
-final class ProductVariantPriceProvider implements ProductVariantPriceProviderInterface
+final class ProductVariantLowestPriceCalculator implements ProductVariantLowestPriceCalculatorInterface
```

Comment:
Why it's a calculator now? IMO provider was fine because we don't calculate the lowest price here. We just fetch it from channel pricing.

Reply:
You could say it's [due to convention](https://github.com/Sylius/Sylius/blob/1.13/src/Sylius/Component/Core/Calculator/ProductVariantPriceCalculator.php) since the price calculators already present in sylius don't calculate anything either. Still, merging it into the core would be easier when their naming/style is virtually the same.